x/protorev/keeper/rebalance.go

```diff
@@ -59,17 +62,42 @@ func (k Keeper) ConvertProfits(ctx sdk.Context, inputCoin sdk.Coin, profit sdk.I
 		return profit, err
 	}
 
+	var poolI poolmanagertypes.PoolI
+
 	// Get the pool
-	conversionPool, err := k.gammKeeper.GetPoolAndPoke(ctx, conversionPoolID)
+	conversionPool, err := k.poolmanagerKeeper.GetPool(ctx, conversionPoolID)
+	if err != nil {
+		return profit, err
+	}
+
+	// get the poolType
+	if conversionPool.GetType() == poolmanagertypes.Concentrated {
+		validCLPool, ok := conversionPool.(cltypes.ConcentratedPoolExtension)
+		if !ok {
+			return profit, fmt.Errorf("pool is not concentrated liquidity pool")
+		}
+
+		poolI = validCLPool
+	} else {
+		validNonClPool, ok := conversionPool.(poolmanagertypes.PoolI)
+		if !ok {
+			return profit, fmt.Errorf("pool is not valid")
+		}
+
+		poolI = validNonClPool
+	}
+
+	swapModule, err := k.poolmanagerKeeper.GetPoolModule(ctx, conversionPoolID)
 	if err != nil {
```

Comment:
can we have test cases for all the changes included in this PR?

Reply:
most of the functions are already covered, i added tests for `GetHighestPoolLiquidity`